src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -209,6 +218,21 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                     isSolutionAvailable,
                     deferredProjectsData.PackageSpecs,
                     token);
+
+                // TODO: To limit risk, we only publish the event when there is a cross-platform PackageReference
+                // project in the solution. Extending this behavior to all solutions is tracked here:
+                // https://github.com/NuGet/Home/issues/4478
+#if !VS14
+                if (projects.OfType<CpsPackageReferenceProject>().Any() &&
+                    !string.IsNullOrEmpty(_dependencyGraphProjectCacheHash))
```

Comment:
>_dependencyGraphProjectCacheHash [](start = 42, length = 32)

why don't we still fire an event if this is the first restore (which i think is if this hash is null or empty)